### PR TITLE
Persist cluster Id when running server in-memory

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Persists Cluster ID between dev server in-memory runs

relies on https://github.com/temporalio/temporal/pull/3883

## Why?

consistent Cluster ID between runs

## Checklist

1. Closes 

2. How was this tested:

manually, WIP unit tests

3. Any docs updates needed?

### CODE REVIEW

File: server/commands.go [hunk 1]
```diff
@@ -274,6 +276,20 @@ func NewServerCommands(defaultCfg *sconfig.Config) []*cli.Command {
 				}
 				if c.String(common.FlagDBPath) == "" {
 					opts = append(opts, WithPersistenceDisabled())
+
+					if clusterCfg, err := cliconfig.NewConfig("temporalio", "cluster"); err == nil {
```
Comment (bergundy) [inline server/commands.go:280]: Maybe call this:

```suggestion
					if clusterCfg, err := cliconfig.NewConfig("temporalio", "server"); err == nil {
```

Comment (bergundy) [reply server/commands.go:280]: So we can add more server configs there if we want.

Comment (bergundy) [reply server/commands.go:280]: Oh and it could be better to have the same structure as the server config.

File: server/commands.go [hunk 2]
```diff
@@ -274,6 +276,20 @@ func NewServerCommands(defaultCfg *sconfig.Config) []*cli.Command {
 				}
 				if c.String(common.FlagDBPath) == "" {
 					opts = append(opts, WithPersistenceDisabled())
+
+					if clusterCfg, err := cliconfig.NewConfig("temporalio", "temporal-version-info"); err == nil {
```
Comment (bergundy) [inline server/commands.go:280]: Please add some comments in this block to explain that we're explicitly ignoring errors.

File: server/commands.go (hunk 1 above)
Comment (feedmeapples) [reply server/commands.go]: i'll look into reusing server's config structure

Comment (bergundy) [reply server/commands.go]: On second thought, having a separate file just for this is better, now that you changed the name I'm okay with the PR as is.

**Review by bergundy — APPROVED** [review]
Had one minor comment.
Otherwise LGTM.

File: server/commands.go (hunk 2 above)
Comment (bergundy) [inline server/commands.go:280]: nit:

```suggestion
					if clusterCfg, err := cliconfig.NewConfig("temporalio", "version-info"); err == nil {
```

**Pull request merged**